plug.vim

```diff
@@ -968,6 +973,364 @@ while 1 " Without TCO, Vim stack is bound to explode
 endwhile
 endfunction
 
+function! s:update_python()
+python << EOF
+import datetime
+import functools
+import os
+import Queue
+import re
+import shlex
+import shutil
+import subprocess
+import tempfile
+import threading as thr
+import time
+import traceback
+import vim
+
+# Global constants, do not modify
+G_IS_WIN = vim.eval('s:is_win') == '1'
+G_MAC_GUI = vim.eval('s:mac_gui') == '1'
+G_PULL = vim.eval('s:update.pull') == '1'
+G_MAX_Y = int(vim.eval('winheight(".")'))
+G_RETRIES = int(vim.eval('get(g:, "plug_retries", 2)')) + 1
+G_TIMEOUT = int(vim.eval('get(g:, "plug_timeout", 60)'))
+G_PROGRESS = vim.eval('s:progress_opt(1)')
+
+class CmdTimedOut(Exception):
+  pass
+class CmdFailed(Exception):
+  pass
+class InvalidURI(Exception):
+  pass
+class Action(object):
+  INSTALL, UPDATE, ERROR, DONE = ['+', '*', 'x', '-']
+
+class GLog(object):
```

Comment:
@starcraftman Yes, it would be useful at this stage. But at some point, it will be stable enough that it's no longer necessary (there hasn't been any bug reports on Ruby installer for quite a while) and I'd prefer to have smaller code base.